runtime/oti/VMHelpers.hpp

```diff
@@ -1364,6 +1364,32 @@ class VM_VMHelpers
 			break;
 		}
 	}
+
+	/**
+	 * @brief Notify JIT upon the first modification of a final field, a stack frame should be build before calling this method to make the stack walkable
+	 * @param currentThread
+	 * @param fieldClass The declaring class of the final field
+	 */
+	static VMINLINE void
+	reportFinalFieldModified(J9VMThread* currentThread, J9Class* fieldClass)
+	{
+		if (J9_ARE_ANY_BITS_SET(currentThread->javaVM->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_OSR_SAFE_POINT)) {
```

Comment:
Please flip these two checks.

Reply:
Also, probably worth using a local for vm and vmFuncs (declared in the appropriate inner scopes).